pkg/knit/cmd/k8s/podres.go

```diff
@@ -197,3 +199,73 @@ func getTopologyInfo(topologyInfo *kubeletpodresourcesv1.TopologyInfo) *Topology
 		Nodes: numaNodes,
 	}
 }
+
+// GetV1Client returns a client for the PodResourcesLister grpc service
+// This was copied from kubelet sources as per the comment there:
+//
+// Quote:
+// Note: Consumers of the pod resources API should not be importing this package.
+// They should copy paste the function in their project.
+func GetV1Client(socket string, connectionTimeout time.Duration, maxMsgSize int) (kubeletpodresourcesv1.PodResourcesListerClient, *grpc.ClientConn, error) {
+	addr, dialer, err := GetAddressAndDialer(socket)
+	if err != nil {
+		return nil, nil, err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), connectionTimeout)
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, addr, grpc.WithInsecure(), grpc.WithContextDialer(dialer), grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(maxMsgSize)))
+	if err != nil {
+		return nil, nil, fmt.Errorf("error dialing socket %s: %v", socket, err)
+	}
+	return kubeletpodresourcesv1.NewPodResourcesListerClient(conn), conn, nil
+}
+
+// GetAddressAndDialer returns the address parsed from the given endpoint and a context dialer.
+func GetAddressAndDialer(endpoint string) (string, func(ctx context.Context, addr string) (net.Conn, error), error) {
```

Comment:
Why do we need this to be public?  In case of kubelet source code this function was in `utils` package so made sense to have it public but since in our case the consumer (GetV1Client) is in the same package making it private would make more sense to me.

Reply:
Fixed.